ksql-cli/src/main/java/io/confluent/ksql/cli/Cli.java

```diff
@@ -505,10 +505,10 @@ private void handlePrintedTopic(String printTopic)
         try {
           topicPrintFuture.get();
         } catch (CancellationException exception) {
+          topicResponse.getResponse().close();
```

Comment:
@big-andy-coates had the comment on the previous comment about this change resulting in topic printing only being closed on cancellation. Your response was that the `topicStreamScanner` being created in a try-with-resources block and thus auto closing. But that close doesn't apply to the `topicResonse`.

Regardless, I am not sure that there is any way to stop 'print topics' outside of a CTRL-C, in effect making cancellation the only way to close the response object. 

@big-andy-coates WDYT?

Reply:
The API doc of `Scanner` mentioned:

> When a Scanner is closed, it will close its input source if the source implements the Closeable interface.

The input source of `topicStreamScanner` is `InputStream` which also implements `Closeable`. So I think that close will also apply to the input source (`topicResponse.getResponse()`).